rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala

```diff
@@ -45,13 +44,28 @@ object RholangCLI {
     }
   }
 
+  private def reader(fileName: String): FileReader = new FileReader(fileName)
+  private def lexer(fileReader: Reader): Yylex     = new Yylex(fileReader)
+  private def parser(lexer: Yylex): parser         = new parser(lexer, lexer.getSymbolFactory())
+
+  private def printTask(normalizedTerm: Par): Task[Unit] =
+    Task {
+      PrettyPrinter.prettyPrint(normalizedTerm)
+      print("\n> ")
+    }
+
   private def repl =
     for (ln <- Source.stdin.getLines) {
-      print(buildNormalizedTerm(new StringReader(ln)).get.toString)
-      print("> ")
+      if (ln.isEmpty) {
+        print("> ")
+      } else {
+        val normalizedTerm = buildNormalizedTerm(new StringReader(ln)).get
+        val evaluator      = printTask(normalizedTerm)
+        evaluator.runAsync
```

Comment:
It seems like this doesn't force ordering between subsequent lines.
But I guess that's not necessary with Rholang.

Reply:
Yeah the REPL is like a par itself :p

Reply:
But I guess if we wanted to have a feature to auto print the current state after each input, it might be confusing to see results get printed out, out of order. We could make it such that if you manage to sneak in two inputs in a row, it only auto prints out once everything reduces.